src/Form/Field/LayoutField.php

```diff
@@ -115,6 +115,33 @@ public function routes(): array
 		return $routes;
 	}
 
+	protected function setDefault($default = null)
+	{
+		// set id for layouts, columns and blocks within layout if not exists
+		if (is_array($default) === true) {
+			$default = array_map(function ($layout) {
+				$layout['id'] ??= Str::uuid();
+
+				// set columns id within layout
+				$layout['columns'] = array_map(function ($column) {
+					$column['id'] ??= Str::uuid();
+
+					// set blocks id within column
+					$column['blocks'] = array_map(function ($block) {
+						$block['id'] ??= Str::uuid();
+						return $block;
+					}, $column['blocks'] ?? []);
+
+					return $column;
+				}, $layout['columns'] ?? []);
+
+				return $layout;
+			}, $default);
```

Comment:
At least barely shorter: 😅

```suggestion
			array_walk($default, function (&$layout) {
				$layout['id'] ??= Str::uuid();

				// set columns id within layout
				if (isset($layout['columns']) === true) {
					array_walk($layout['columns'], function (&$column) {
						$column['id'] ??= Str::uuid();

						// set blocks id within column
						if (isset($layout['blocks']) === true) {
							array_walk($column['blocks'], function (&$block) {
								$block['id'] ??= Str::uuid();
							});
						}
					});
				}
			});
```

Reply:
Thanks! There was a small variable typo in the last if condition. I pushed it manually 👍